### ISSUE TITLE
Have a feeling sources.list is missing the main repositories

### ISSUE THREAD

**Issue by madebydavid** [pablobm/w2c3-livecd#15]
I was playing around with a fresh install (2019-04-02^mirror-screen-shortcut) and wanted to install the htop package.

As root, I ran `apt-get update` and `apt-cache search htop` but nothing was found.

I checked the `/etc/apt/sources.list` file and it has the following (after some commented out lines):

```
deb http://security.debian.org/debian-security stretch/updates main contrib non-free
deb-src http://security.debian.org/debian-security stretch/updates main contrib non-free
```

I updated this to add:

```
deb http://deb.debian.org/debian/ stretch main contrib non-free
deb-src http://deb.debian.org/debian/ stretch main contrib non-free

deb http://deb.debian.org/debian/ stretch-updates main contrib non-free
deb-src http://deb.debian.org/debian/ stretch-updates main contrib non-free
```

After this as root I ran `apt-get update` and `apt-cache search htop` and I could then locate and install the package.

Happy to fix this one myself if you could point out the steps for me @pablobm 

Thank you!

**Comment by pablobm** [comment]
I think I have unsuccessfully tried to fix this in the past, but I'm not 100% sure. I'll explain what I remember.

In general, files found at `config/includes.chroot` are copied onto the final image. So in theory adding your own `sources.list` at `config/includes.chroot/etc/apt` should be the thing to do. However, I think when I tried in the past this was overwritten by the installer, leaving us with the configuration you have found.

If this is the case, then perhaps the file should be kept elsewhere, and copied to the right place from one of the "hook" scripts at `config/hooks/normal` (possibly including a call to `apt update`).

When I say "kept elsewhere", this should also be somewhere in `config/includes.chroot`, as you won't have access to files outside that hierarchy when hook scripts are running.

Does that help? I can try give it a go too; let me know.

**Comment by madebydavid** [comment]
Ok, this has been an epic journey.

I tried your route first with a file called `1020-fix-sources-list.hook.chroot`

```bash
#!/usr/bin/env sh
set -e

# This is to resolve https://github.com/pablobm/w2c3-livecd/issues/15 - we
# manually set the sources.list to the debian defaults - without this it seems
# to onlu have the security repo included

cat << "EOF" > /etc/apt/sources.list
deb http://security.debian.org/debian-security stretch/updates main contrib non-free
deb-src http://security.debian.org/debian-security stretch/updates main contrib non-free

deb http://deb.debian.org/debian/ stretch main contrib non-free
deb-src http://deb.debian.org/debian/ stretch main contrib non-free

deb http://deb.debian.org/debian/ stretch-updates main contrib non-free
deb-src http://deb.debian.org/debian/ stretch-updates main contrib non-free
EOF

apt-get update
```
This did not work.

My first thought was because of the `.chroot` extension it meant it would apply the sources to the chroot and not the image itself so I tried calling the file `9030-fix-sources-list.hook.binary` and setting it as executable. Again, no luck.

I then consulted the [manual here](https://live-team.pages.debian.net/live-manual/html/live-manual/customizing-package-installation.en.html#379) which suggested adding the sources file itself in `config/archives` - I added it as `config/archives/w2c3.list.binary`. This seemed to make a difference, however when installing on my test laptop I would get a red screen at the point of installing grub saying "grub-pc failed to install into /target/" - I pressed Alt+F4 to get the error and it looked like this:

![image](https://user-images.githubusercontent.com/5401249/69015608-86831e00-098d-11ea-8d1b-2b2dd4e27fa6.png)

It looks like it was failing because it was trying to connect to the internet during the install (it is strange that it fails at the grub stage rather than earlier).

So, next I consulted the manual again and saw the "Distribution mirrors at runtime" [option](https://live-team.pages.debian.net/live-manual/html/live-manual/customizing-package-installation.en.html#376) which suggested adding a param to the `ld config` command - I did it like this in the `auto/config` file:

```
#!/bin/sh

set -e

lb config noauto \
	--archive-areas 'main contrib non-free' \
	--debian-installer live \
	--debian-installer-preseedfile preseed.cfg \
	--mirror-binary http://ftp.uk.debian.org/debian/ \
	"${@}"
```

Again, no luck.

Perhaps we can write a hook file which adds an init script which sets the sources on boot?

**Comment by pablobm** [comment]
Yeah, your last suggestion is something I was thinking of too. It's terrible but what can you do.

Let me quickly ask on the mailing list, to see if there's another way (I should have done this long ago).

**Comment by pablobm** [comment]
Got an answer from the mailing list: https://lists.debian.org/debian-live/2019/11/msg00019.html

Does that help?

**Comment by madebydavid** [comment]
Humm, they suggest a `list.chroot` file in `config/archives` - I tried with the `list.binary` in there and had the issue in the photo above. Will give this option a go!

In that repo they link to, the `list.chroot` file is [mostly commented out](https://github.com/SebKuzminsky/linuxcnc-wheezy-iso/blob/44816be9ac712d88a422e7d71602b1381f3cf7ef/images/xfce-desktop/config/archives/linuxcnc.list.chroot), but it has a `preseed/late_command` in the preseed file [here](https://github.com/SebKuzminsky/linuxcnc-wheezy-iso/blob/b69db30d90a8e2b591d2a0f8289668f0914a589c/images/xfce-desktop/config/debian-installer/preseed.cfg#L26) which seems to change the sources.list. Maybe that's the secret ingredient?

I will try both and get back to you. Thank you!

**Comment by madebydavid** [comment]
Second option works have opened a [PR](https://github.com/pablobm/w2c3-livecd/pull/18)

**Comment by madebydavid** [comment]
Resolved in #18
